rascal-lsp/src/main/rascal/lang/rascal/lsp/Outline.rsc

```diff
@@ -31,38 +31,58 @@ import String;
 import ParseTree;
 import lang::rascal::\syntax::Rascal;
 import util::LanguageServer;
+import util::ErrorRecovery;
 
 list[DocumentSymbol] outlineRascalModule(start[Module] \mod) {
     m= \mod.top;
-    children = [];
-
-    top-down-break visit (m) {
-        case (Declaration) `<Tags _> <Visibility _> <Type t> <{Variable ","}+ vars>;`:
-            children += [symbol(clean("<v.name>"), variable(), v@\loc, detail="variable <t> <v>") | v <- vars];
-
-        case (Declaration) `<Tags _> <Visibility _> anno <Type t> <Type ot>@<Name name>;`:
-            children +=  [symbol(clean("<name>"), field(), t@\loc, detail="anno <t> <ot>")];
 
-        case (Declaration) `<Tags _> <Visibility _> alias <UserType u> = <Type al>;`:
-            children += [symbol(clean("<u.name>"), struct(), u@\loc, detail="<u> = <al>")];
+    if (!(m has header) || hasErrors(m.header)) {
+        return [];
+    }
 
-        case (Declaration) `<Tags _> <Visibility _> tag <Kind k> <Name name> on <{Type ","}+ ts>;`:
-            children += [symbol(clean("<name>"), \key(), name@\loc, detail="tag <k> <name> on <ts>")];
+    children = [];
 
-        case (Declaration) `<Tags _> <Visibility _> data <UserType u> <CommonKeywordParameters kws>;`: {
-            kwlist = [symbol(".<k.name>", property(), k@\loc, detail="<k.\type>") | kws is present, KeywordFormal k <- kws.keywordFormalList];
-            children += [symbol("<u.name>", struct(), u@\loc, detail="data <u> <kws>", children=kwlist)];
+    top-down-break visit (m) {
+        case decl: (Declaration) `<Tags _> <Visibility _> <Type t> <{Variable ","}+ vars>;`:
+            if (!hasErrors(decl)) {
+                children += [symbol(clean("<v.name>"), variable(), v@\loc, detail="variable <t> <v>") | v <- vars, !hasErrors(v)];
+            }
+
+        case decl: (Declaration) `<Tags _> <Visibility _> anno <Type t> <Type ot>@<Name name>;`:
```

Comment:
once the match succeeds we now this node is not in error, so we don't need to check `hasErrors` for that.. but...